### ISSUE TITLE
Release on itch.io ?

### ISSUE THREAD

**Issue by belovachap** [belovachap/SubBoomUnity#12]
It's pretty easy to create an account, upload whatever Unity builds for Mac/Windows/Linux as a zip file and get some people to look at your game using https://itch.io/ If you want you can host our executables on your account :) I can show you what I've learned about Unity builds whenever we meet up next.

**Comment by spicewooddev** [comment]
i already have an itch.io account actually! we could totally upload it on there once we finish it. i was honestly thinking we can try to make a WebGL build of the game-- being able to play the game on your browser. im not sure how that would affect the stats text in the main menu though.

and making a checklist of things we still need to fix would be a good idea so we're both on the same page of what needs to be done.

**Comment by belovachap** [comment]
It seems to run pretty decent in the browser (admittedly I'm currently on a desktop computer with a good graphics card so I'll have to try it out on my laptop which just has a CPU) https://www.chapmanshoop.com/subboom_web/ The score doesn't persist across web page reloads but it does work within the game session. If you click the "Quit" button the application also just kind of "freezes" which is less than ideal but also I'm not too worried about it.

If you want we can also go about setting up a small web server for you so you can host the game there as well.
It looks like you can get a tiny droplet in the Digital Ocean for $4 a month https://www.digitalocean.com/pricing/droplets
A domain name might run you $10-$20 a year https://www.namecheap.com/
I'm not entirely sure what the right tools for SSH'ing out to the server are in Windows anymore. I think once-upon-a-time we used "PUTTY" https://www.ssh.com/academy/ssh/putty/windows/install
Maybe we could used VirtualBox and have an Ubuntu virtual machine on your computer :)

But it's really easy to install Apache and have a web server out-of-the-box. I also think HTML/CSS/Javascript are pretty neat :)

**Comment by spicewooddev** [comment]
Right now, if you were to play for longer than a minute, it does get pretty laggy in Unity and on the website it uses quite a bit of memory (and i have a gaming laptop). This is mostly due to the amount of bubbles spawning and being constantly updated during the game. We may want to add bubbles just for the projectiles to reduce the amount and/or use a particle system instead, but not sure how to go about the ladder option if that's possible.

Some other future options for optimization we can try is object pooling, where instead of constantly creating and destroying game objects, which uses more CPU, we can instead activate and deactivate game objects instead.

And for servers, I think it would be best to use WebGL for Unity and upload it on itch.io instead. That way it can have its own website, and we won't have to worry about spending any money. Plus, I don't have a job so I'd rather not spend any money, and I wouldn't want you to either for this project.

**Comment by belovachap** [comment]
Yeah, we can try to use fewer bubbles to get better performance and less memory usage :)

I didn't know you could host the web app on itch.io, that sounds perfect! Free options are great :)

**Comment by belovachap** [comment]
It might be interesting to have the WebGL version up on itch.io and then try to sell the stand alone builds for Mac/Windows/Linux there as well. Maybe some people will like it enough to want to play it offline on their computer :) I feel like it might be $5 worth of fun :D

**Comment by spicewooddev** [comment]
that's a good idea! i wonder if there's a way to automatically split the money from sales so that we both can get a share without dealing with managing the sales.

**Comment by belovachap** [comment]
> that's a good idea! i wonder if there's a way to automatically split the money from sales so that we both can get a share without dealing with managing the sales.

I googled it and looked around in the itch.io settings and it seems like there isn't a way to setup revenue splitting between developers. It's okay if it just goes to your account, I'm mostly just curious to know if anyone (and if anyone how many ones) actually buy the game.

**Comment by spicewooddev** [comment]
maybe 1 or 2 people lol. our game isn't super eye-catching when it comes to visuals and it's not a lengthy game at all.
